File: twidere/src/androidTest/java/org/mariotaku/twidere/util/TwidereTypeUtilsTest.java
package org.mariotaku.twidere.util;

import org.apache.commons.lang3.reflect.TypeUtils;
import org.junit.Test;
import org.mariotaku.microblog.library.twitter.model.CursorTimestampResponse;
import org.mariotaku.microblog.library.twitter.model.ResponseList;
import org.mariotaku.microblog.library.twitter.model.Status;

import java.util.List;

import static org.junit.Assert.assertEquals;

/**
 * Created by mariotaku on 16/2/15.
 */
public class TwidereTypeUtilsTest {

    @Test
    public void testGetSimpleName() throws Exception {
        assertEquals("CursorTimestampResponse", TwidereTypeUtils.toSimpleName(CursorTimestampResponse.class));
        assertEquals("ResponseList<Status>", TwidereTypeUtils.toSimpleName(TypeUtils.parameterize(ResponseList.class, Status.class)));
        assertEquals("List<List<Object>>", TwidereTypeUtils.toSimpleName(TypeUtils.parameterize(List.class, TypeUtils.parameterize(List.class, Object.class))));
    }
}

File: twidere/src/androidTest/java/org/mariotaku/twidere/util/TwidereTypeUtilsTest.kt
package org.mariotaku.twidere.util

import org.apache.commons.lang3.reflect.TypeUtils
import org.junit.Assert.assertEquals
import org.junit.Test
import org.mariotaku.microblog.library.twitter.model.CursorTimestampResponse
import org.mariotaku.microblog.library.twitter.model.ResponseList
import org.mariotaku.microblog.library.twitter.model.Status

/**
 * Created by mariotaku on 16/2/15.
 */
class TwidereTypeUtilsTest {

    @Test
    @Throws(Exception::class)
    fun testGetSimpleName() {
        assertEquals("CursorTimestampResponse", TwidereTypeUtils.toSimpleName(CursorTimestampResponse::class.java))
        assertEquals("ResponseList<Status>", TwidereTypeUtils.toSimpleName(TypeUtils.parameterize(ResponseList::class.java, Status::class.java)))
        assertEquals("List<List<Object>>", TwidereTypeUtils.toSimpleName(TypeUtils.parameterize(List::class.java, TypeUtils.parameterize(List::class.java, Any::class.java))))
    }
}